make/golang.mk

```diff
@@ -27,37 +27,37 @@ _ := $(if  \
 		$(error must be running Go version ^$(1) - you are running $(shell go version | cut -d" " -f3 | cut -c3-)))
 endef
 
-export GO15VENDOREXPERIMENT=1
-
 # FGT is a utility that exits with 1 whenever any stderr/stdout output is recieved.
 FGT := $(GOPATH)/bin/fgt
 $(FGT):
 	go get github.com/GeertJohan/fgt
 
-# Godep is a tool used to manage Golang dependencies in the style of the Go 1.5
-# vendoring experiment.
-GODEP := $(GOPATH)/bin/godep
-$(GODEP):
-	go get -u github.com/tools/godep
+DEP_VERSION = v0.3.2
 
-# Golint is a tool for linting Golang code for common errors.
-GOLINT := $(GOPATH)/bin/golint
-$(GOLINT):
-	go get github.com/golang/lint/golint
+# Dep is a tool used to manage Golang dependencies. It is the offical vendoring experiment, but
+# not yet the official tool for Golang.
+$(GOPATH)/src/github.com/golang/dep:
+	@git clone --branch $(DEP_VERSION) https://github.com/golang/dep.git $(GOPATH)/src/github.com/golang/dep
+$(GOPATH)/bin/dep: $(GOPATH)/src/github.com/golang/dep
+	@git -C $(GOPATH)/src/github.com/golang/dep checkout tags/$(DEP_VERSION)
+	@go build -o $(GOPATH)/bin/dep github.com/golang/dep/cmd/dep
 
-# golang-vendor-deps installs all dependencies needed for different test cases.
-golang-godep-vendor-deps: $(GODEP)
+golang-dep-vendor-deps: $(GOPATH)/bin/dep
 
-# golang-godep-vendor is a target for saving dependencies with the godep tool
-# to the vendor/ directory. All nested vendor/ directories are deleted as they
-# are not handled well by the Go toolchain.
+# golang-godep-vendor is a target for saving dependencies with the dep tool
+# to the vendor/ directory. All nested vendor/ directories are deleted via
+# the prune command.
 # arg1: pkg path
-define golang-godep-vendor
-$(GODEP) save $(1)
-@# remove any nested vendor directories
-find vendor/ -path '*/vendor' -type d | xargs -IX rm -r X
+define golang-dep-vendor
+dep ensure
+dep prune
```

Comment:
I don't think we need `prune` – `dep ensure` already removes nested vendor directories. prune gets rid of unrecognized/unused packages in your vendor tree.